acl-engine/src/main/java/javasabr/mqtt/acl/engine/model/matcher/dynamic/DynamicTopicMatcher.java

```diff
@@ -0,0 +1,77 @@
+package javasabr.mqtt.acl.engine.model.matcher.dynamic;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import java.util.Map;
+import javasabr.mqtt.acl.engine.model.matcher.TopicMatcher;
+import javasabr.mqtt.base.util.DebugUtils;
+import javasabr.mqtt.model.MqttUser;
+import javasabr.mqtt.model.topic.AbstractTopic;
+import javasabr.mqtt.model.topic.TopicFilter;
+import javasabr.mqtt.model.topic.TopicName;
+import javasabr.mqtt.model.topic.TopicValidator;
+import lombok.AccessLevel;
+import lombok.CustomLog;
+import lombok.experimental.FieldDefaults;
+
+@CustomLog
+@FieldDefaults(level = AccessLevel.PROTECTED, makeFinal = true)
+public abstract class DynamicTopicMatcher<T extends AbstractTopic> implements TopicMatcher {
+
+  static {
+    DebugUtils.registerIncludedFields("originalTopic", "resolvers");
+  }
+  
+  public static TopicMatcher autoBuild(String rawOriginalTopic) {
+    if (TopicValidator.validateTopicName(rawOriginalTopic)) {
+      return new DynamicTopicNameMatcher(TopicName.valueOf(rawOriginalTopic));
+    } else if (TopicValidator.validateTopicFilter(rawOriginalTopic)) {
+      return new DynamicTopicFilterMatcher(TopicFilter.valueOf(rawOriginalTopic));
+    }
+    throw new IllegalArgumentException("Invalid topic:" + rawOriginalTopic);
+  }
+  
+  T originalTopic;
+  TopicSegmentResolver[] resolvers;
+
+  protected DynamicTopicMatcher(T originalTopic) {
+    String[] rawSegments = originalTopic.segments();
```

Comment:
What do you think about the idea of ​​implementing an additional `DynamicTopicFilter` entity in addition to the existing `TopicName` and `TopicFilter`? This will avoid the spread of segment logic beyond the topic entity.